### ISSUE TITLE
ejabberd crashes with mod_http_upload enabled

### ISSUE THREAD

**Issue by mbirth** [processone/ejabberd#1451]
> What version of ejabberd are you using?

16.12-beta1 (from the official Docker image)

> What operating system (version) are you using?

Whatever you used in that Docker image

> How did you install ejabberd (source, package, distribution)?

Docker

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behavior? What was the expected result?

As soon as I enable mod_http_upload in settings like this:

```
-
  port: 5280
  module: ejabberd_http
  ...
  request_handlers:
    "/oauth": ejabberd_oauth
    "/api": mod_http_api
    "/upload": mod_http_upload

....

  mod_http_upload:
    docroot: "/home/p1/cfg/upload"
```

ejabberd crashes on startup:

```
2017-01-07 20:24:35.458 [critical] <0.1003.0>@gen_mod:start_module:162 Problem starting the module mod_http_upload for host <<"localhost">>
 options: [{docroot,<<"/home/p1/cfg/upload">>}]
 error: undef
[{http_uri,parse,["http://localhost"],[]},
 {mod_http_upload,get_proc_name,2,
                  [{file,"src/mod_http_upload.erl"},{line,511}]},
 {mod_http_upload,start,2,[{file,"src/mod_http_upload.erl"},{line,148}]},
 {gen_mod,start_module,3,[{file,"src/gen_mod.erl"},{line,154}]},
 {lists,foreach,2,[{file,"lists.erl"},{line,1338}]},
 {ejabberd_app,start,2,[{file,"src/ejabberd_app.erl"},{line,77}]},
 {application_master,start_it_old,4,
                     [{file,"application_master.erl"},{line,273}]}]
2017-01-07 20:24:35.459 [critical] <0.1003.0>@gen_mod:maybe_halt_ejabberd:170 ejabberd initialization was aborted because a module start failed.
```

Setting `put_url` didn't make a difference.

**Comment by zinid** [comment]
This looks like an image issue. It doesn't include `http_uri` module or something.

**Comment by mbirth** [comment]
The OS seems to be Alpine Linux (https://github.com/processone/docker-ejabberd), so the package would be `erlang-inets` - but it seems you didn't use Alpine's erlang packages but included `erts-8.1` in that `ejabberd.tar.gz` that was installed to `/home/p1/ejabberd`.

**Comment by cromain** [comment]
right, thanks pointing that out

**Comment by cromain** [comment]
fixed by e56c654
our official 17.03 docker images will be published after ejabberd release.
